guides/cookbook.md

```diff
@@ -31,141 +80,177 @@ constructing input streams to read files), which is out of scope for this
 cookbook. Code such as this will be replaced by a method with an empty (but
 implied) implementation.
 
-> *Java*
-> 
-> Import statements for classes internal to the Ion library or to `java.lang`
-> are omitted. Other external classes will either be first referenced by their
-> fully-qualified names, or will be preceded by an import statement. Import
-> statements have global scope.
+## Reading and Writing Ion Data
+
+<script>writeTabs()</script>
+<div class="tabpane Java" markdown="1">
+Implementations of the [`IonReader`][4] and [`IonWriter`][5] interfaces are
+responsible for reading and writing Ion data in both text and binary forms.
 
-* * *
+`IonReader`s and `IonWriter`s may be constructed through builders.
 
-## Reading and Writing Ion Data
+An [`IonReaderBuilder`][16] with the default configuration may be constructed as
+follows. This builder will construct `IonReader` instances which can read
+both text and binary Ion data.
+
+```java
+IonReaderBuilder readerBuilder = IonReaderBuilder.standard();
+```
+
+An [`IonTextWriterBuilder`][6] with the default configuration may be constructed
+as follows. This builder will construct `IonWriter` instances which output data
+in the text Ion format.
+
+```java
+IonTextWriterBuilder textWriterBuilder = IonTextWriterBuilder.standard();
+```
+
+To construct a builder that constructs `IonWriter` instances which output data
+in the binary Ion format, use an [`IonBinaryWriterBuilder`][15].
+
+```java
+IonBinaryWriterBuilder binaryWriterBuilder = IonBinaryWriterBuilder.standard();
+```
+
+Each of the aforementioned builders may be used to construct multiple `IonReader`
+or `IonWriter` instances with the same configuration.
+
+Consider the following text Ion data, which has been materialized as a Java
+String.
+
+```java
+String helloWorld = "{ hello:\"world\" }";
+```
+
+An `IonReader` for this data may be constructed as follows.
+
+```java
+IonReader reader = readerBuilder.build(helloWorld);
+```
+
+Reading the data requires leveraging the `IonReader`'s APIs.
+
+```java
+void readHelloWorld() {
+    reader.next();                            // position the reader at the first value, a struct
+    reader.stepIn();                          // step into the struct
+    reader.next();                            // position the reader at the first value in the struct
+    String hello = reader.getFieldName();     // retrieve the current value's field name
```

Comment:
Naming the variable the same characters as the string is confusing for me.

Reply:
agreed; renamed variables to `fieldName` and `value` in the Java and JavaScript code for this example